module/VuFind/src/VuFind/Controller/AbstractSolrSearch.php

```diff
@@ -136,7 +146,23 @@ protected function processAdvancedFacets($facetList, $searchObject = false,
             // to a flat array according to the hierarchy
             if (in_array($facet, $hierarchicalFacets)) {
                 $tmpList = $list['list'];
-                $facetHelper->sortFacetList($tmpList, true);
+
+                $sort = $hierarchicalFacetsSortOptions['*'] ?? 'top';
```

Comment:
A proposed refactor that I think might be a little less confusing:

<pre>
                $sort = $hierarchicalFacetsSortOptions[$facet]
                    ?? $hierarchicalFacetsSortOptions['*'] ?? 'top';
                switch ($sort) {
                case 'count':
                    // Do nothing; count order is Solr's default behavior.
                    break;
                case 'all':
                    $facetHelper->sortFacetList($tmpList, false);
                    break;
                case 'top':
                default:
                    $facetHelper->sortFacetList($tmpList, true);
                }
</pre>

This avoids repeatedly redefining the $sort variable, and I think doing the sorting inline in the switch instead of creating an extra flag is more clear... but please let me know if I've messed anything up with this proposal. :-)

Reply:
I agree, the refactored version is easier to read.